packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -908,6 +922,162 @@ class _LargeTitleNavigationBarSliverDelegate
+  }
+}
+
+class _RenderLargeTitle extends RenderShiftedBox {
+  _RenderLargeTitle(
+    super.child, {
+    required BoxConstraints navBarConstraints,
+    required double maxExtent,
+    required TextDirection textDirection,
+  })  : _navBarConstraints = navBarConstraints,
+        _maxExtent = maxExtent,
+        _textDirection = textDirection;
+
+  BoxConstraints get navBarConstraints => _navBarConstraints;
+  BoxConstraints _navBarConstraints;
+  set navBarConstraints(BoxConstraints value) {
+    if (_navBarConstraints == value) {
+      return;
+    }
+    _navBarConstraints = value;
+
+    markNeedsLayout();
+  }
+
+  double get maxExtent => _maxExtent;
+  double _maxExtent;
+  set maxExtent(double value) {
+    if (_maxExtent == value) {
+      return;
+    }
+    _maxExtent = value;
+
+    markNeedsLayout();
+  }
+
+  TextDirection get textDirection => _textDirection;
+  TextDirection _textDirection;
+  set textDirection(TextDirection value) {
+    if (_textDirection == value) {
+      return;
+    }
+    _textDirection = value;
+
+    markNeedsLayout();
+  }
+
+  late Matrix4 _transform;
+
+  @override
+  void performLayout() {
+    child?.layout(constraints.loosen(), parentUsesSize: true);
+
+    // Maximum scale lets us prevent the large title
+    // from getting clipped when its width is greater than
+    // the navigation bar's max width constraint.
+    final double maxScale = child!.size.width != 0.0
+      ? clampDouble((navBarConstraints.maxWidth - _kNavBarEdgePadding) / child!.size.width, 1.0, 1.1)
+      : 1.1;
+
+    // This scale is estimated from the settings app in iOS 16.
+    // The large title scales linearly from 1.0 up to 1.1 magnification.
+    // The `navBarConstraints.maxHeight` value is the height of the nav bar,
+    // and `maxExtent` is the default large title height the nav bar snaps back to.
+    // The difference between the two heights is used to scale the title.
+    final double scale = clampDouble(1.0 + (navBarConstraints.maxHeight - maxExtent) / maxExtent * 0.12, 1.0, maxScale);
+
+    size = constraints.constrainDimensions(
+      double.infinity, 
+      child!.size.height * scale,
+    );
+
+    final Alignment resolvedAlignment =
+        AlignmentDirectional.bottomStart.resolve(textDirection);
+    final Offset translation = resolvedAlignment.alongSize(size);
+
+    final BoxParentData childParentData = child!.parentData! as BoxParentData;
+    childParentData.offset =
+        resolvedAlignment.alongOffset(size - child!.size as Offset);
+
+    final Matrix4 resultMatrix = Matrix4.identity();
+    resultMatrix
+      ..translate(translation.dx, translation.dy)
+      ..scale(scale, scale)
+      ..translate(-translation.dx, -translation.dy);
+
+    _transform = resultMatrix;
+  }
+
+  @override
+  void paint(PaintingContext context, Offset offset) {
+    layer = context.pushTransform(
+      needsCompositing,
+      offset,
+      _transform,
+      super.paint,
```

Comment:
super.paint already accounted for the translation part of the transform I think?